packages/medusa/src/types/price-list.ts

```diff
@@ -9,14 +9,18 @@ import {
   ValidateNested,
 } from "class-validator"
 import { PriceList } from "../models/price-list"
-import { DateComparisonOperator } from "./common"
+import { DateComparisonOperator, FindConfig } from "./common"
 import { XorConstraint } from "./validators/xor"
 
 export enum PriceListType {
   SALE = "sale",
   OVERRIDE = "override",
 }
 
+export type PriceListListQuery = FindConfig<PriceList> & {
+  where: Partial<{ -readonly [key in keyof PriceList]: PriceList[key] }>
```

Comment:
@pKorsholm I would suggest that you do something like 
```ts
where: Partial<Writable<PriceList>>
```
The `Writable` type will come along that [PR](https://github.com/medusajs/medusa/pull/1344/files#diff-82b54536d7ac72b91dcd04c4c65e660b310080eaa5077e7bad551f563aaf1d51R10) but even if you add it as well it will be a small conflict or no conflict :)

wdyt?

Reply:
Yeah, I think it's a good idea to just use that. I have stolen that and the result from the build query you added and used that instead. 

Do you think we should make a specialized type for each time we use it in a repository, or should we use the generalized type from `common.ts`. wdyt?

Reply:
I think using the generalised type from common will suffice 😀